secboot/secboot_sb_test.go

```diff
@@ -890,115 +887,86 @@ func (s *secbootSuite) TestSealKey(c *C) {
-			},
-			{
-				Source: sb_efi.Firmware,
-				Image:  sb_efi.FileImage(mockBF[1].Path),
-				Next:   cdk,
-			},
-		}
+		sequences2 := sb_efi.NewImageLoadSequences().Append(
+			sb_efi.NewImageLoadActivity(
+				sb_efi.NewFileImage(mockBF[0].Path),
+				sb_efi.Firmware,
+			).Loads(cdk...),
+			sb_efi.NewImageLoadActivity(
+				sb_efi.NewFileImage(mockBF[1].Path),
+				sb_efi.Firmware,
+			).Loads(cdk...),
+		)
 
 		tpm, restore := mockSbTPMConnection(c, tc.tpmErr)
 		defer restore()
 
 		// mock adding EFI secure boot policy profile
-		var pcrProfile *sb_tpm2.PCRProtectionProfile
-		addEFISbPolicyCalls := 0
-		restore = secboot.MockSbEfiAddSecureBootPolicyProfile(func(profile *sb_tpm2.PCRProtectionProfile, params *sb_efi.SecureBootPolicyProfileParams) error {
-			addEFISbPolicyCalls++
-			pcrProfile = profile
-			c.Assert(params.PCRAlgorithm, Equals, tpm2.HashAlgorithmSHA256)
-			switch addEFISbPolicyCalls {
-			case 1:
-				c.Assert(params.LoadSequences, DeepEquals, sequences1)
-			case 2:
-				c.Assert(params.LoadSequences, DeepEquals, sequences2)
-			default:
-				c.Error("AddSecureBootPolicyProfile shouldn't be called a third time")
-			}
-			return tc.addEFISbPolicyErr
-		})
-		defer restore()
 
-		// mock adding EFI boot manager profile
-		addEFIBootManagerCalls := 0
-		restore = secboot.MockSbEfiAddBootManagerProfile(func(profile *sb_tpm2.PCRProtectionProfile, params *sb_efi.BootManagerProfileParams) error {
-			addEFIBootManagerCalls++
-			c.Assert(profile, Equals, pcrProfile)
-			c.Assert(params.PCRAlgorithm, Equals, tpm2.HashAlgorithmSHA256)
-			switch addEFISbPolicyCalls {
+		addPCRProfileCalls := 0
+		restore = secboot.MockSbEfiAddPCRProfile(func(pcrAlg tpm2.HashAlgorithmId, branch *sb_tpm2.PCRProtectionProfileBranch, loadSequences *sb_efi.ImageLoadSequences, options ...sb_efi.PCRProfileOption) error {
+			addPCRProfileCalls++
+			c.Assert(pcrAlg, Equals, tpm2.HashAlgorithmSHA256)
+			switch addPCRProfileCalls {
 			case 1:
-				c.Assert(params.LoadSequences, DeepEquals, sequences1)
+				c.Assert(loadSequences, DeepEquals, sequences1)
 			case 2:
-				c.Assert(params.LoadSequences, DeepEquals, sequences2)
+				c.Assert(loadSequences, DeepEquals, sequences2)
 			default:
-				c.Error("AddBootManagerProfile shouldn't be called a third time")
+				c.Error("AddPCRProfile shouldn't be called a third time")
 			}
-			return tc.addEFIBootManagerErr
+			return tc.addPCRProfileErr
 		})
 		defer restore()
 
 		// mock adding systemd EFI stub profile
 		addSystemdEfiStubCalls := 0
-		restore = secboot.MockSbEfiAddSystemdStubProfile(func(profile *sb_tpm2.PCRProtectionProfile, params *sb_efi.SystemdStubProfileParams) error {
+		restore = secboot.MockSbEfiAddSystemdStubProfile(func(profile *sb_tpm2.PCRProtectionProfileBranch, params *sb_efi.SystemdStubProfileParams) error {
 			addSystemdEfiStubCalls++
-			c.Assert(profile, Equals, pcrProfile)
+			// XXX: not sure what we are testing with this:
+			//c.Assert(profile, Equals, pcrProfile)
```

Comment:
This was from mvo. I am also uncertain what needs to be tested.

Reply:
I think I got it. We were just checking we were using only one profile here. Fixed it.